test/metabase/api/downloads_exports_test.clj

```diff
@@ -0,0 +1,184 @@
+(ns metabase.api.downloads-exports-test
+  "Tests for the various Download and Export features in the app.
+
+  TODO: Add tests to cover the many places data can be exported or downloaded:
+  - Question direct downloads
+  - Unsaved question direct downloads
+  - Dashcard direct downloads
+  - Shared Question downloads
+  - Static Embedding Dashboard/dashcard downloads
+  - Dashboard Subscription Attachments
+  - Alert attachments"
+  (:require
+   [clojure.data.csv :as csv]
+   [clojure.set :as set]
+   [clojure.test :refer :all]
+   [metabase.test :as mt]
+   [metabase.test.data.interface :as tx]))
+
+(def ^:private pivot-base-rows
+  (for [a ["AA" "AB" "AC" "AD"]
+        b ["BA" "BB" "BC" "BD"]
+        c ["CA" "CB" "CC" "CD"]
+        d ["DA" "DB" "DC" "DD"]]
+    [a b c d 1]))
+
+(tx/defdataset pivot-base-data
+  [["pivot-base"
+    [{:field-name "Cat A", :base-type :type/Text}
+     {:field-name "Cat B", :base-type :type/Text}
+     {:field-name "Cat C", :base-type :type/Text}
+     {:field-name "Cat D", :base-type :type/Text}
+     {:field-name "measure", :base-type :type/Quantity}]
+    pivot-base-rows]])
+
+;; The Pivot Table Download/export test can be a bit confusing. I've kept a 'see pivot result' function in a comment at the end of this ns
+;; If you eval/run that in your repl, you should be able to see the results (It's not too many rows so should print acceptably)
+;; If you need to add assertions or fix up this test, that may be a helpful function to run!
+(deftest pivot-export-test
+  []
+  (mt/dataset pivot-base-data
+    (mt/with-temp [:model/Card {card-id :id}
```

Comment:
Going forward for tests that run queries it's preferable to use a mocked metadata provider (so you don't need to hit the app DB at all) instead of `with-temp`, you could do something like 

```clj
(qp.store/with-metadata-provider (lib.tu/mock-metadata-provider
                                  (lib.metabase.jvm/application-database-metadata-provider (mt/id))
                                  {:cards [{:id                     1
                                            :name                   "Card 1"
                                            :display                :pivot
                                            :visualization-settings {}
                                            :dataset-query          {}}]})
  ...)
```

here. Don't feel obligated to fix this  test tho, this is just a suggestion for next time

Reply:
Nevermind I take it back, that wouldn't work since you're hitting a Card API endpoint. If you ran the query outside of the REST API that would have been the way to do it